python/TestHarness/TestHarness.py

```diff
@@ -698,7 +707,7 @@ def handleTestResult(self, tester, output, result, start=0, end=0, add_to_table=
     def cleanup(self):
         # Print the results table again if a bunch of output was spewed to the screen between
         # tests as they were running
-        if self.options.verbose or (self.num_failed != 0 and not self.options.quiet):
+        if (self.options.verbose or (self.num_failed != 0 and not self.options.quiet)) and not self.options.dry_run:
             print '\n\nFinal Test Results:\n' + ('-' * (TERM_COLS-1))
             for (test, output, result, timing, start, end) in sorted(self.test_table, key=lambda x: x[2], reverse=True):
                 print printResult(test, result, timing, start, end, self.options)
```

Comment:
How about:
```bash
-------------------------------------------------------------------------------------------------------------
Processed 1366 tests in 0.8 seconds
1307 would run, 46 would be skipped
```
I removed the odd looking predictability of failed tests:
```bash
./run_tests --dry-run -e
-------------------------------------------------------------------------------------------------------------
Processed 1366 tests in 0.8 seconds
1307 would run, 46 will be skipped, 13 are sure to fail
```
haha, because even though accurate, it is darned odd to see. Especially when there are normally 0 tests sure to fail.